lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -488,16 +490,22 @@ def safe_partman_create_parent(
       end
     end
 
+    validated_table = PgHaMigrations::PartmanTable.from_table_name(table)
+    validated_template_table = template_table ? PgHaMigrations::PartmanTable.from_table_name(template_table) : nil
+
     create_parent_options = {
-      parent_table: _fully_qualified_table_name_for_partman(table),
-      template_table: template_table ? _fully_qualified_table_name_for_partman(template_table) : nil,
+      parent_table: validated_table.fully_qualified_name,
+      template_table: validated_template_table&.fully_qualified_name,
       control: partition_key,
-      type: "native",
       interval: interval,
       premake: premake,
       start_partition: formatted_start_partition,
     }.compact
 
+    if PgHaMigrations.partman.major_version < 5
+      create_parent_options[:type] = "native"
+    end
```

Comment:
As mentioned in the PR description, this is the only conditional logic required to get this working (in the sense that partman doesn't raise violent errors)